### DISCUSSION TITLE
Multiple Temp Directory usage logic?

### DISCUSSION THREAD

**Discussion by michaelc95** [ericaltendorf/plotman#567] [Q&A]
Hiya

Amazing program! Thank you for sharing it with the community! 

If I have 6 temp directories for instance that are big enough to hold 2 temp plots each (and I tell Plotman this via overrides)
What is Plotman's default behavior when distributing jobs to these directories?  

Does it do this           
P1 to T1                              
P2 to T2                                
P3 to T3                                             
P4 to T4                                            
P5 to T5                                            
P6 to T6                                                
Before it turns around and uses         
P7 to T1                                            
and so on...                                      
 
OR

Does it do this 
P1 to T1 
P2 to T1  
P3 to T2
P4 to T2
P5 to T3
P6 to T3 
All the way up to
P12 to T6
Before it turns around and uses
P13 to T1

I would prefer it follow the first set of logic as that would have a greater chance of 2 plots on 1 drive being in 
totally separate phases based on my stagger thus having less IO impact then 2 plots in phase 1 pounding the drive. 
I am plotting on HD's by the way.. 24x 600gb sas10k drives so the goal is to evenly distribute 48 parallel temp plots IO over them
as best I can with a 30 minute stagger. Once I understand plotman's logic I can figure out if I need to raid 0 drives together or not. 

Thanks,
Mike

**Comment by cyperbg** [comment]
See if this helps: https://i.imgur.com/el4POzG.png

I believe it is the former way. Each disk gets a plot and then it repeats.

**Comment by JoshuaEstes** [comment]
The logic is more like the first. However it seems that your concern is multiple plots running in phase one. using `tmpdir_stagger_phase_major` set to `2` and `tmpdir_stagger_phase_minor` set to `1` and `tmpdir_stagger_phase_limit` set to `1` basically means that ONLY 1 plot can in the previous phases. If you set `tmpdir_stagger_phase_limit` to `2`, it just means that there can be a MAX of 2 plots in phase 2:0 or less instead of the 1.

It seems that as plotman is ready to kick off another job it goes down the list until it finds a tmp dir that it can use. If you are in `plotman interactive` you will either see "OK" or "- -" for "ready" when it's listing all your tmp dirs. It will NOT kick off a job in any of the tmp dirs that aren't ready.
